packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java

```diff
@@ -368,13 +368,7 @@ public void clearFocus() {
 
   @Override
   public boolean requestFocus(int direction, Rect previouslyFocusedRect) {
-    // This is a no-op so that when the OS calls requestFocus(), nothing will happen. ReactEditText
-    // is a controlled component, which means its focus is controlled by JS, with two exceptions:
-    // autofocus when it's attached to the window, and responding to accessibility events. In both
-    // of these cases, we call requestFocusInternal() directly.
-    // Always return true if we are already focused. This is used by android in certain places,
-    // such as text selection.
-    return isFocused();
+    return requestFocusInternal();
```

Comment:
I'm not super comfortable with this one. We should understand the decision made here better and build manual test cases to make sure everything works as expected before and after.

Reply:
Building additional test cases would be great. 

As far as I understand the reasoning behind this is that requestFocus was not used for mobile as there it makes sense to control focus from JS.

But for TVs currently you are unable to focus TextInput. I've been using a Touchable wrapper to onFocus call searchRef.focus() so this feels like a hack. (Maybe there is a better solution?)

If you take a look at requestFocusInternal it handles all cases like softKeyboardFocus etc. so I think we are covered there.

But I agree that this small looking change might be a breaking  change for some apps as this will change the behavior to work properly 😅

Reply:
@okwasniewski @yusufyildirim I think the right way to handle this change would be to put it behind a `ReactFeatureFlags` property (we did this for developers who needed to get key down events from the keyboard/DPAD).